src/services/componentServices.v2.js

```diff
@@ -0,0 +1,323 @@
+// TEMPORARY FILE TO STORE VERSION 2 OF SOME FUNCTIONS IN componentServices.js
+// FUNCTIONS IN THIS FILE WILL BE MOVED INTO THE ORIGINAL FILE ONCE REQUIRED
+
+const config = require('./config.json');
+const constants = config.constants;
+const facilityServices = require(config.paths.facilityServices);
+const dbContext = require('../repository/DbContext');
+const Component = require(config.paths.componentModel);
+
+/**
+ * Creates a new component.
+ * @param {Component} component must be a component object.
+ * @param {String} tempFacilityName temporary facility name which component will be assigned to, 
+ *  if it's null, then the component's facility will be assigned with facility id inside the component object
+ * @returns {Boolean} true if successful, false otherwise.
+ */
+ const addComponent = async ({ component, tempFacilityName = null }) => {
+    let resStatus = "OK";
+
+    const knex = dbContext.getKnexObject();
+
+    let newValue = convertComponentInputValue(component.value);
+
+    let newIsChild = 0;
+    let newParentId = null;
+
+    if (component.isChild) {
+        newIsChild = 1;
+        newParentId = component.parentId;
+    }
+
+    let costCalcReturn = await calculateComponentCost(component.cost, component.regionId);
+
+    if(costCalcReturn === "Cost exceeding treasury amount"){
+        resStatus = costCalcReturn;
+        return resStatus
+    }
+
+    if (tempFacilityName != null) {
+        const mapKey = -1;
```

Comment:
Why's the mapKey -1?

Reply:
It's arbitrary because the mapKey could literally be anything, any string or any number.

Reply:
How about we make it `null` or `undefined` or something to make it more clear that it's a lack of value?

Reply:
It's not lack of value though, the value can be anything. Variable `mapKey` here works as the key of the facility dictionary. However, I tested using `null` value and it still works, so I'll change it to that.